internal/gatewayapi/helpers.go

```diff
@@ -159,7 +159,7 @@ func GetReferencedListeners(parentRef v1beta1.ParentReference, gateways []*Gatew
 
 		// The parentRef may be to the entire Gateway, or to a specific listener.
 		for _, listener := range gateway.listeners {
-			if parentRef.SectionName == nil || *parentRef.SectionName == listener.Name {
+			if (parentRef.SectionName == nil || *parentRef.SectionName == listener.Name) && (parentRef.Port == nil || *parentRef.Port == listener.Port) {
```

Comment:
guessing this check was added to make it tighter based on API spec https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.ParentReference

Reply:
Yeah, I think `port` was added to the ParentReference struct more recently so we didn't have support for it yet.

Reply:
Ah thanks for confirming